File: app/src/main/java/id/ac/stiki/doleno/absenin/domain/RegisterEvent.java
package id.ac.stiki.doleno.absenin.domain;

import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentReference;

import id.ac.stiki.doleno.absenin.data.entity.EventParticipant;

public interface RegisterEvent {
    Task<DocumentReference> execute(EventParticipant eventParticipant);
}

File: app/src/main/java/id/ac/stiki/doleno/absenin/domain/RegisterEvent.kt
package id.ac.stiki.doleno.absenin.domain

import com.google.android.gms.tasks.Task
import id.ac.stiki.doleno.absenin.data.entity.Event
import id.ac.stiki.doleno.absenin.data.entity.User

interface RegisterEvent {
    suspend fun execute(event: Event, user: User): Task<Void>
}